python/cudf/cudf/api/extensions/accessor.py

```diff
@@ -0,0 +1,35 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+
+class AccessorManager:
```

Comment:
This looks a lot like a [`cached_property`](https://docs.python.org/dev/library/functools.html?highlight=s#functools.cached_property) with an additional `name` attribute.

We have a cached_property implementation [here](https://docs.python.org/dev/library/functools.html?highlight=s#functools.cached_property).

I wonder if there's any scope for code reuse (and in the future, just using `functools.cached_property`).

Reply:
Apologies, looks like I linked to the wrong place: https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/utils/utils.py#L314

Reply:
There is one thing that may prevent from reusing the code: `CachedProperty` uses `setattr`, but for `DataFrame`, `__setattr__` is overridden, so we have to use `object.__setattr__` instead.

Reply:
No worries. I think a separate descriptor from cached_property is totally fine

Reply:
Or actually, `Pandas` has this implementation already, is it ok to reuse their code? https://github.com/pandas-dev/pandas/blob/2a7d3326dee660824a8433ffd01065f8ac37f7d6/pandas/core/accessor.py#L159-L193

Reply:
Yes, reusing Pandas's implementation sounds like a good approach to me.

Reply:
To be clear, are you referring to `import`-ing their `CachedAccessor`, or copying the code?

importing is of course totally OK.

In the latter case, we may have to include the Pandas license (? @kkraus14 ) 

https://github.com/pandas-dev/pandas/blob/master/LICENSE

Reply:
Importing. Taking this one step further. Can we also import [_register_accessor](https://github.com/pandas-dev/pandas/blob/2a7d3326dee660824a8433ffd01065f8ac37f7d6/pandas/core/accessor.py#L196-L282) ?

Reply:
Underscore attributes are understood to be "non-public", so if possible we should avoid doing that